include/onnxruntime/core/framework/tensor_shape.h

```diff
@@ -30,7 +60,7 @@ class TensorShape {
   TensorShape& operator=(TensorShape&& other) noexcept;
 
   TensorShape(gsl::span<const int64_t> dims);
-  TensorShape(const std::vector<int64_t>& dims) : TensorShape(gsl::make_span(dims)) {}
+  TensorShape(const TensorShapeVector& dims) : TensorShape(gsl::make_span(dims)) {}
```

Comment:
Can we reinterpret_cast a TensorShapeVector to TensorShape without copy? If not, why do we still need two types? If we know the vector will be used as tensor shape,  why don't use a TensorShape object directly ?

Reply:
TensorShapeVector is intended to be used instead of std::vector<int64_t> to build a shape. TensorShape is an immutable class which could have TensorShapeVector as a member, but could not serve as a base since TensorShape can also store a ptr to an external buffer (See from existing buffer).